go-controller/pkg/ovn/master.go

```diff
@@ -117,7 +118,12 @@ func (oc *Controller) Start(nodeName string, wg *sync.WaitGroup) error {
 		return err
 	}
 
-	go leaderElector.Run(context.Background())
+	wg.Add(1)
+	go func() {
+		leaderElector.Run(ctx)
+		klog.Infof("Stopped leader election")
```

Comment:
@dcbw I don't see this `Stopped leader election` in the logs, and I suspect this code is never run because:

```go
func (le *LeaderElector) Run(ctx context.Context) {
	defer runtime.HandleCrash()
	defer func() {
		le.config.Callbacks.OnStoppedLeading()
	}()
```

So we call `os.Exit(1)` first.

```go
			OnStoppedLeading: func() {
				klog.Infof("No longer leader; exiting")
				os.Exit(1)
			}
```

And never hit the `klog.Infof("Stopped leader election")` or the `wg.Done()`

The lock is still released, but this is dead code it seems.

```
I0409 02:27:42.941540       1 ovnkube.go:121] Received signal terminated. Shutting down
I0409 02:27:42.941694       1 services_controller.go:164] Shutting down controller ovn-lb-controller
I0409 02:27:42.941794       1 reflector.go:225] Stopping reflector *v1.Service (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.941875       1 reflector.go:225] Stopping reflector *v1.Pod (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.941944       1 reflector.go:225] Stopping reflector *v1.EgressIP (0s) from github.com/openshift/ovn-kubernetes/go-controller/pkg/crd/egressip/v1/apis/informers/externalversions/factory.go:117
I0409 02:27:42.942193       1 reflector.go:225] Stopping reflector *v1beta1.CustomResourceDefinition (0s) from k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions/factory.go:117
I0409 02:27:42.942240       1 reflector.go:225] Stopping reflector *v1.NetworkPolicy (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.942287       1 reflector.go:225] Stopping reflector *v1.Node (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.942326       1 reflector.go:225] Stopping reflector *v1.Service (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.942358       1 reflector.go:225] Stopping reflector *v1.Endpoints (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.942406       1 reflector.go:225] Stopping reflector *v1.Namespace (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.942445       1 reflector.go:225] Stopping reflector *v1beta1.EndpointSlice (0s) from k8s.io/client-go/informers/factory.go:134
I0409 02:27:42.959646       1 master.go:106] No longer leader; exiting
```